plugin/src/main/java/com/denizenscript/denizen/scripts/commands/entity/InvisibleCommand.java

```diff
@@ -66,185 +64,62 @@ public InvisibleCommand() {
-            return;
-        }
-        ElementTag state = scriptEntry.getElement("state");
-        List<PlayerTag> forPlayers = scriptEntry.argForPrefixList("for", PlayerTag.class, true);
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), target, state, db("for", forPlayers));
-        }
-        switch (state.asEnum(Action.class)) {
-            case TRUE:
-            case FALSE:
-                if (forPlayers == null) {
-                    setInvisible(target, state.asBoolean());
-                }
-                else {
-                    boolean invisible = state.asBoolean();
-                    for (PlayerTag player : forPlayers) {
-                        setInvisibleForPlayer(target, player, invisible);
-                    }
-                }
-                break;
-            case TOGGLE:
-                if (forPlayers == null) {
-                    setInvisible(target, !isInvisible(target.getBukkitEntity(), null, false));
-                }
-                else {
-                    for (PlayerTag player : forPlayers) {
-                        setInvisibleForPlayer(target, player, !isInvisible(target.getBukkitEntity(), player.getUUID(), false));
-                    }
-                }
-                break;
-            case RESET:
-                HashMap<UUID, Boolean> playerMap = invisibleEntities.get(target.getUUID());
-                if (playerMap == null) {
-                    return;
-                }
-                HashSet<UUID> playersToUpdate = new HashSet<>();
-                if (forPlayers == null) {
-                    playersToUpdate.addAll(playerMap.keySet());
-                    invisibleEntities.remove(target.getUUID());
-                }
-                else {
-                    for (PlayerTag player : forPlayers) {
-                        playerMap.remove(player.getUUID());
-                        playersToUpdate.add(player.getUUID());
-                    }
-                    if (playerMap.isEmpty()) {
-                        invisibleEntities.remove(target.getUUID());
-                    }
-                }
-                if (!playersToUpdate.isEmpty()) {
-                    for (Player player : NMSHandler.entityHelper.getPlayersThatSee(target.getBukkitEntity())) {
-                        if (playersToUpdate.contains(player.getUniqueId())) {
-                            NMSHandler.packetHelper.sendEntityMetadataFlagsUpdate(player, target.getBukkitEntity());
-                        }
-                    }
-                }
-                break;
-        }
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("target") @ArgLinear @ArgDefaultNull @ArgSubType(EntityTag.class) List<EntityTag> targets,
+                                   @ArgName("state") @ArgDefaultText("TRUE") EntityMetadataCommandHelper.Action action,

```

Comment:
lowercase is preferred, even for enums (as they're internally lowercased by the enum handlers anyway)

Reply:
Ah I see, should be lowercased now